vignettes/stan-help.Rmd

```diff
@@ -0,0 +1,403 @@
+
+```r
+# In a fresh R session run the following in your R console.
+install.packages("cmdstanr", repos = c("https://mc-stan.org/r-packages/", getOption("repos")))
+
+```
+
+To ensure that your toolchain instllation occured sucessfully, run the following code in your R terminal:
+
+```r
+check_cmdstan_toolchain()
+```
+
+This function will report back if the toolchain is available and set-up as follows:
+
+```r
+#> The C++ toolchain required for CmdStan is setup properly!
+```
+
+If you do not get this message, return to [the installation instuctions](#toolchain) and ensure that all steps were followed.
+
+Assuming you have the the toolchain installed, you can install CmdStanR.
+
+```r
+cmdstanr::install_cmdstan(cores = 2)
+```
+
+# Epinowcast modeling
+
+## Installation
+
+As described in the project README, you can install the `"{epinowcast}"` package within R.
+
+```r
+# Stable version from:
+install.packages("epinowcast", repos = "https://epinowcast.r-universe.dev")
+
+# Development version from:
+remotes::install_github("epinowcast/epinowcast", dependencies = TRUE)
+```
+
+## Running your first model
+
+The first time you fit a model with `{epinowcast}` will be **slow** compared to future fits!
+This is normal as all of the Stan code is being compiled for the first time.
+Note that the compiled Stan binaries will live in your user package directory (or wherever you have defined your package libraries).
+Because of this package architecture
+
+- Each time you install a new version of `{epinowcast}` you will experience this longer compilation time the first time you fit a model.
+- If you upgrade your version of CmdStan, you will likely need to recompile your models. This may occur automatically, or necessitate re-installing `{epinowcast}`.
+
+
+## Setting `enw_fit_opts`
+
+The `enw_fit_opts` allows you to set several critical components for the Stan computational engine.
+These parameters are passed to Stan during the model fitting process and have an influence on computation time and quality of the model fit.
+Knowing when and if the defaults need to be changed is an important part of the Bayesian workflow.[@gelmanBayesianWorkflow2020]
+All of the parameters are passed to the `sampling` function available from [`{cmdstanr}`](https://mc-stan.org/cmdstanr/reference/model-method-sample.html).
+
+# Failures modes
+
+In this section we will discuss the most common issues when working with `{epinowcast}` specifically and CmdStan/ Bayesian inference more generally.
+The solutions to these issues typically require you to investigate your [sampler settings](#samplerParams), [model settings](#modelParams), or [interogate your data](#modelData) further.
+
+## Sampler settings {.samplerParams}
+
+`{epinowcast}` provides users the ability to pass arguments directly to CmdStan which can improve model run time and produce better inferences.
+These parameters can be passed the `enw_fit_opts` function.
+
+### `threads_per_chain` and `chains`
```

Comment:
I would suggest splitting these two into separate subsections. You do a good job differentiating chains from threads in the text and mark threads as purely performance-focused. But, they have such different implications that I think it's confusing to talk about them together.